### PR TITLE
[CORE-3407] Add sharding host parser

### PR DESCRIPTION
**Pull request by dubinin-dmytro**

### CODE REVIEW

File: src/Queue/QlessConnector.php [hunk 1]
```diff
@@ -13,22 +13,106 @@
...
+
+        if (!$value) {
+            return [];
+        }
+
+        $values = explode(self::CONFIG_SEPARATOR, $value);
+
+        return array_map(function ($v) {
+            return trim($v) ?: null;
+        }, $values);
```
Comment (osadchyi-s) [inline src/Queue/QlessConnector.php:100]:
```suggestion
        return array_map('trim', $values);
```

File: src/Queue/QlessConnector.php [hunk 2]
```diff
@@ -13,22 +13,106 @@
...
+        $redisConnection = Arr::get($config,
+            self::REDIS_CONNECTION_CONFIG_KEY,
+            self::DEFAULT_CONNECTION_CONFIG);
+
+        $qlessConfig = Config::get(self::CONFIG_PATH_PREFIX . $redisConnection, []);
 
-         $redisConfig = Config::get('database.redis.' . $redisConnection, []);
+        if ($this->hasSharding($qlessConfig)) {
+            $shardingConfigs = $this->getShardingConfigs($qlessConfig);
+            $qlessConfig = $shardingConfigs[rand(0, count($shardingConfigs) - 1)];
```
Comment (osadchyi-s) [inline src/Queue/QlessConnector.php:49]: If we have one process (worker) - qless will be work only with one connection?

Comment (osadchyi-s) [inline src/Queue/QlessConnector.php:49]:
```suggestion
            $qlessConfig = $shardingConfigs[array_rand($shardingConfigs)];
```

Comment (osadchyi-s) [reply src/Queue/QlessConnector.php:49]: We need to push in random Redis, but workers need to read all redises

File: src/Queue/QlessConnector.php [hunk 3]
```diff
@@ -13,22 +13,106 @@
...
+
+    private function extractConfigValues(string $key, array $connectionConfig): array
+    {
+        $value = $connectionConfig[$key] ?? null;
+
+        if (!$value) {
+            return [];
+        }
+
+        $values = explode(self::CONFIG_SEPARATOR, $value);
```
Comment (osadchyi-s) [inline src/Queue/QlessConnector.php:96]: I think we can just leave variants: old and array

File: src/Queue/QlessConnector.php (hunk 1 above)
Comment (dubinin-dmytro) [reply src/Queue/QlessConnector.php:100]: тут еще фильтр пустые значения. они заменяются на null

File: config/database.php [hunk 4]
```diff
@@ -8,5 +8,17 @@
...
+            'host' => env('REDIS_HOST', '127.0.0.1'),
+            'password' => env('REDIS_PASSWORD', null),
+            'port' => env('REDIS_PORT', 6379),
+            'database' => 0,
+        ],
+        'qless2' => [
+            'host' => env('REDIS_HOST', '127.0.0.1'),
+            'password' => env('REDIS_PASSWORD', null),
+            'port' => env('REDIS_PORT', 6379),
+            'database' => 0,
```
Comment (osadchyi-s) [inline config/database.php:21]: Вкажи для прикладу різні database. Ну або ще одну константу додати

File: src/Queue/QlessQueue.php [hunk 5]
```diff
@@ -37,14 +32,17 @@ class QlessQueue extends Queue implements QueueContract
      */
     private $config;
 
+    /** @var QlessConnectionHandler */
```
Comment (osadchyi-s) [inline src/Queue/QlessQueue.php:35]: It is not necessary, but better when an order of attributes and arguments of a constructor is the same

**Review by osadchyi-s — COMMENTED** [review]
Great work

**Review by SkrebtsovHeorhii — APPROVED** [review]

**Pull request merged**